lib/request-processor.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+var _ = require('underscore')
+  , parser = require('./parser')
+  , Subscriber = require('./subscriber')
+  , messageCreator = require('./message-creator');
+
+var subscriptionHandlers = {
+  subscribe: function (phoneNumber, movie) {
+    Subscriber.find(phoneNumber)
+      .then(function(subscriber) {
+        var movies = addMovie(subscriber, movie);
+        Subscriber.create(phoneNumber, movies);
+      });
+  },
+
+  unsub: function (phoneNumber, movie) {
+    Subscriber.find(phoneNumber)
+      .then(function(subscriber) {
+        if (subscriber) {
+          var movies = deleteMovie(subscriber, movie);
+          if (_.isEmpty(movies)) {
+            Subscriber.delete(subscriber.id);
+          } else {
+            Subscriber.create(phoneNumber, movies);
+          }
+        }
+      });
+  }
+};
+
+var addMovie = function(subscriber, movie) {
+  if (_.isUndefined(subscriber)) {
```

Comment:
Short answer: Tell don't ask.

A more interesting answer: I wan't this method to have the logic to know if there isn't a subscriber, if that is the case (not in this method of course) I'll create a new subscriber with a subscription for the movie passed as an argument.

If a subscriber already exists, then I'll replace the subscriber with the a new one which will contains the previous movie they have been subscribed and the new one.

This makes sense to you? 😅